core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -880,15 +883,24 @@ public UnnestAnalysis getUnnest(Unnest node)
     public void addTableColumnReferences(AccessControl accessControl, Identity identity, Multimap<QualifiedObjectName, String> tableColumnMap)
     {
         AccessControlInfo accessControlInfo = new AccessControlInfo(accessControl, identity);
-        Map<QualifiedObjectName, Set<String>> references = tableColumnReferences.computeIfAbsent(accessControlInfo, k -> new LinkedHashMap<>());
+        Map<QualifiedObjectName, Set<String>> referencesInStatement = tableColumnReferencesInStatements.computeIfAbsent(accessControlInfo, k -> new LinkedHashMap<>());
+        Map<QualifiedObjectName, Set<String>> rowFilterReferences = tableColumnReferencesInRowFilter.computeIfAbsent(accessControlInfo, k -> new LinkedHashMap<>());
         tableColumnMap.asMap()
-                .forEach((key, value) -> references.computeIfAbsent(key, k -> new HashSet<>()).addAll(value));
+                .forEach((key, value) -> {
+                    //Allow to create a rowfilter on a not accessibile column
+                    if (hasRowFilter(key, identity.getUser())) {
```

Comment:
`Analysis` is just a giant collection. It should not do any logic. The logic should be moved to the caller.